pyproject.toml

```diff
@@ -55,6 +52,7 @@ pyproject-pre-commit = "^0.1.4"
 pyfakefs = "^5.3.1"
 black = "^24.4.0"
 isort = "^5.13.2"
+deptry = "^0.16.2"
```

Comment:
Just wondering whether we need this to be declared as a dependency on an ongoing basis? I guess if we want to run it regularly then it couldn't hurt. Do you know if it has many dependencies itself?

Reply:
In theory we could just install and run it periodically as needed. Might be worth removing if it results in a bunch of dependabot requests.
Alternatively it might be an easy one to just include in pre-commit hooks to just double check you haven't removed the last instance of a dependency.